drools-wb-screens/drools-wb-test-scenario-editor/drools-wb-test-scenario-editor-backend/src/main/java/org/drools/workbench/screens/testscenario/backend/server/ScenarioLoader.java

```diff
@@ -71,13 +71,14 @@
         FileExtensionFilter fileExtensionFilter = new FileExtensionFilter(".scenario");
 
         // Get list of immediate children
-        final DirectoryStream<org.uberfire.java.nio.file.Path> directoryStream = ioService.newDirectoryStream(pPath);
-        for (final org.uberfire.java.nio.file.Path p : directoryStream) {
-            if (filter.accept(p) && fileExtensionFilter.accept(p)) {
-                if (Files.isRegularFile(p)) {
-                    items.add(Paths.convert(p));
-                } else if (Files.isDirectory(p)) {
-                    items.add(Paths.convert(p));
+        try (final DirectoryStream<org.uberfire.java.nio.file.Path> directoryStream = ioService.newDirectoryStream(pPath)) {
+            for (final org.uberfire.java.nio.file.Path p : directoryStream) {
+                if (filter.accept(p) && fileExtensionFilter.accept(p)) {
```

Comment:
Seeing this code, I can't help but dislike all this nesting. How about:

    if (!filter.accept(p) || !fileExtensionFilter.accept(p)) {
        continue;
    } else if (Files.isRegularFile(p) || Files.isDirectory(p)) {
        items.add(Paths.convert(p));
    }

Alternatively, for a more fun approach, all the filters and conditions could be merged into a single chained `Predicate<Path>`:

    Predicate<Path> acceptableFileName = filter::accept;
    Predicate<Path> acceptablePath = acceptableFilename.and(fileExtensionFilter::accept)
        .and(Files::isRegularFile)
        .and(Files::isDirectory);
    ...
    if (acceptablePath.test(p)) {
       items.add(Paths.convert(p);
    }

Anyway... probably not worth making the change right here right now. Just food for thought.

Reply:
Thanks Lukas, I filed a task for myself https://issues.jboss.org/browse/BAQE-501